src/services/goals.js

```diff
@@ -457,8 +448,79 @@ async function removeObjectives(currentObjectiveIds) {
   });
 }
 
+export async function removeRemovedRecipientsGoals(removedRecipientIds, report) {
+  if (!removedRecipientIds) {
+    return null;
+  }
+
+  const goals = await Goal.findAll({
+    attributes: ['id'],
+    where: {
+      grantId: removedRecipientIds,
+    },
+    include: [
+      {
+        model: ActivityReport,
+        as: 'activityReports',
+        required: true,
+        where: {
+          id: report.id,
+        },
+      },
+    ],
+  });
+
+  const goalIds = goals.map((goal) => goal.id);
+
+  await ActivityReportGoal.destroy({
+    where: {
+      goalId: goalIds,
+      activityReportId: report.id,
+    },
+  });
+
+  const objectives = await Objective.findAll({
+    attributes: ['id'],
+    where: {
+      goalId: goalIds,
+    },
+    include: [
+      {
+        model: ActivityReportObjective,
+        as: 'activityReportObjectives',
+        required: true,
+        where: {
+          activityReportId: report.id,
+        },
+      },
+    ],
+  });
+
+  const objectiveIds = objectives.map((objective) => objective.id);
+
+  await ActivityReportObjective.destroy({
+    where: {
+      objectiveId: objectiveIds,
+      activityReportId: report.id,
+    },
+  });
+
+  await Objective.destroy({
+    where: {
+      id: objectiveIds,
+    },
+  });
+
+  return Goal.destroy({
+    where: {
+      id: goalIds,
+      onApprovedAR: false,
+    },
+  });
```

Comment:
we can't directly delete. we need to check the objective and goal to make sure there is no other reports its used on.

Reply:
Think this should be good now